Python/pylifecycle.c

```diff
@@ -2199,6 +2198,13 @@ is_valid_fd(int fd)
     _Py_END_SUPPRESS_IPH
 
     return (fd2 >= 0);
+#elif defined(MS_WINDOWS)
+    HANDLE hfile;
+    _Py_BEGIN_SUPPRESS_IPH
+    hfile = (HANDLE)_get_osfhandle(fd);
+    _Py_END_SUPPRESS_IPH
+    return (hfile != INVALID_HANDLE_VALUE &&
+              GetFileType(hfile) != FILE_TYPE_UNKNOWN);
```

Comment:
I have a personal preference for:

```suggestion
    return (hfile != INVALID_HANDLE_VALUE
            && GetFileType(hfile) != FILE_TYPE_UNKNOWN);
```